overlord/servicestate/servicestate.go

```diff
@@ -49,18 +49,55 @@ type Instruction struct {
 
 type ServiceActionConflictError struct{ error }
 
+type ResolvedAppInfo struct {
+	snap.AppInfo
+	ExplicitlyRequested bool
+}
+
+func NewResolvedAppInfo(app *snap.AppInfo) *ResolvedAppInfo {
+	resolvedAppInfo := new(ResolvedAppInfo)
+	resolvedAppInfo.AppInfo = *app
+	return resolvedAppInfo
+}
+
+type ResolvedAppInfos []*ResolvedAppInfo
+
+func NewResolvedAppInfos(appInfos []*snap.AppInfo) ResolvedAppInfos {
+	resolvedAppInfos := make(ResolvedAppInfos, 0, len(appInfos))
+	for _, app := range appInfos {
+		resolvedAppInfos = append(resolvedAppInfos, NewResolvedAppInfo(app))
+	}
+	return resolvedAppInfos
+}
+
+func (a ResolvedAppInfos) Len() int      { return len(a) }
```

Comment:
This will be used with sort.Sort(), and should probably have a distinct and more informative name, eg.

```
type ResolvedAppInfosByAppName ResolvedAppInfos
```